app/controllers/rx_controller.rb

```diff
@@ -4,13 +4,21 @@
 class RxController < ApplicationController
   include ActionController::Serialization
 
-  skip_before_action :authenticate
+  before_action :authorize_rx
   before_action :authenticate_client
 
   protected
 
   def client
-    @client ||= Rx::Client.new(session: { user_id: ENV['MHV_USER_ID'] })
+    @client ||= Rx::Client.new(session: { user_id: mhv_correlation_id })
+  end
+
+  def authorize_rx
+    mhv_correlation_id || raise(Common::Exceptions::Unauthorized)
```

Comment:
From SerializableError class, you can see that the Exceptions are capable of supporting all of the following:

```
      attribute :title, String
      attribute :detail, String
      attribute :id, Integer
      attribute :href, String
      attribute :code, String
      attribute :source, String
      # attribute :links, Array[String] # commenting this out for now as it complicates things
      attribute :status, String
      attribute :meta, String
```